omnibus/config/software/libbcc_compat.patch

```diff
@@ -0,0 +1,74 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7bd0f3b..2c5eb36 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -13,6 +13,8 @@ enable_testing()
+ if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/src/cc/libbpf/src)
+   execute_process(COMMAND git submodule update --init --recursive
+                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
++  execute_process(COMMAND sed -i -E s/\\<reallocarray\\>/my_reallocarray/ scripts/check-reallocarray.sh include/tools/libc_compat.h src/libbpf.c
++                  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/cc/libbpf)
+ endif()
+
+ include(cmake/GetGitRevisionDescription.cmake)
+diff --git a/src/cc/CMakeLists.txt b/src/cc/CMakeLists.txt
+index c53c542..8244f77 100644
+--- a/src/cc/CMakeLists.txt
++++ b/src/cc/CMakeLists.txt
+@@ -46,9 +46,11 @@ set(libbpf_uapi libbpf/include/uapi/linux})
+
+ add_library(bpf-static STATIC libbpf.c perf_reader.c ${libbpf_sources})
+ set_target_properties(bpf-static PROPERTIES OUTPUT_NAME bcc_bpf)
++set_target_properties(bpf-static PROPERTIES COMPILE_DEFINITIONS COMPAT_NEED_REALLOCARRAY)
+ add_library(bpf-shared SHARED libbpf.c perf_reader.c ${libbpf_sources})
+ set_target_properties(bpf-shared PROPERTIES VERSION ${REVISION_LAST} SOVERSION 0)
+ set_target_properties(bpf-shared PROPERTIES OUTPUT_NAME bcc_bpf)
++set_target_properties(bpf-shared PROPERTIES COMPILE_DEFINITIONS COMPAT_NEED_REALLOCARRAY)
+
+ set(bcc_common_sources bcc_common.cc bpf_module.cc bcc_btf.cc exported_files.cc)
+ if (${LLVM_PACKAGE_VERSION} VERSION_EQUAL 6 OR ${LLVM_PACKAGE_VERSION} VERSION_GREATER 6)
+@@ -71,7 +73,7 @@ set(bcc_api_headers bcc_common.h bpf_module.h bcc_exception.h bcc_syms.h bcc_pro
+ if(ENABLE_CLANG_JIT)
+ add_library(bcc-shared SHARED
+   link_all.cc ${bcc_common_sources} ${bcc_table_sources} ${bcc_sym_sources}
+-  ${bcc_util_sources})
++  ${bcc_util_sources} wrapper.c)
+ set_target_properties(bcc-shared PROPERTIES VERSION ${REVISION_LAST} SOVERSION 0)
+ set_target_properties(bcc-shared PROPERTIES OUTPUT_NAME bcc)
+
+@@ -133,7 +135,7 @@ endif()
+ add_subdirectory(frontends)
+
+ # Link against LLVM libraries
+-target_link_libraries(bcc-shared ${bcc_common_libs_for_s})
++target_link_libraries(bcc-shared ${bcc_common_libs_for_s} -Wl,--wrap=exp -Wl,--wrap=log -Wl,--wrap=pow -Wl,--wrap=exp2 -Wl,--wrap=log2)
+ target_link_libraries(bcc-static ${bcc_common_libs_for_a} bcc-loader-static)
+ set(bcc-lua-static ${bcc-lua-static} ${bcc_common_libs_for_lua})
+
+diff --git a/src/cc/wrapper.c b/src/cc/wrapper.c
+new file mode 100644
+index 0000000..29e9e6b
+--- /dev/null
++++ b/src/cc/wrapper.c
+@@ -0,0 +1,21 @@
```

Comment:
I would add some comments here (mostly with what you have in your PR description). I'm afraid that coming back to this without comment in few months/years is going to be painful

Reply:
I added the whole description (description of the PR which is also the commit message) on top of the patch.